expo/expo
# Fix React Native Appearance/useColorScheme support for SDK 38

# Why

`useColorScheme` has two problems right now:
- It doesn't update on Android, but it does on iOS. Related issue: https://github.com/facebook/react-native/issues/28823 - it requires we override `onConfigurationChanged` in our activity.
- It doesn't respect the configuration when users force a particular color scheme via `userInterfaceStyle` in app.json.

# How

- Added the `onConfigurationChanged` method as required, verified that this fixes `useColorScheme` from `react-native`
- Copy over relevant code from `ExpoAppearanceModule` to give priority to the current activity local night mode configuration (which is what we use to enforce a specific scheme when `userInterfaceStyle` is set in app.json). This required bumping appcompat to 1.1.0 in ReactAndroid.

# TODO

- [ ] The changes to BaseExperienceActivity will apply to SDK38 just fine, but the ReactAndroid changes will not. These need to be applied to SDK38 too, and I'm not familiar with the process for this on Android.

# Test Plan

- Use the tabs template
- If TODO hasn't been resolved, set the sdkVersion to UNVERSIONED
- Set `userInterfaceStyle` to `light` in app.json and verify that it's light when you use dark mode, and nothing happens when you toggle between modes while it's open. Same for dark. Verify automatic works as expected.
